src/frontend/src/icp-eth/components/tokens/AddTokenByNetwork.svelte

```diff
@@ -0,0 +1,40 @@
+<script lang="ts">
+	import { i18n } from '$lib/stores/i18n.store';
+	import { Dropdown, DropdownItem } from '@dfinity/gix-components';
+	import { networksMainnets } from '$lib/derived/networks.derived';
+	import { ETHEREUM_NETWORK_SYMBOL, ICP_NETWORK_SYMBOL } from '$env/networks.env';
+	import IcAddTokenForm from '$icp/components/tokens/IcAddTokenForm.svelte';
+	import AddTokenForm from '$eth/components/tokens/AddTokenForm.svelte';
+	import { networkId } from '$lib/derived/network.derived';
+	import { isNullish } from '@dfinity/utils';
+
+	export let dropdownNetwork: string | undefined;
+	export let tokenData: Record<string, string>;
+
+	let ledgerCanisterId: string = '';
+	let indexCanisterId: string = '';
+	let contractAddress: string = '';
+
+	$: tokenData = {
+		ledgerCanisterId,
+		indexCanisterId,
+		contractAddress
+	};
+</script>
+
+{#if isNullish($networkId)}
```

Comment:
Assuming the user can manage all tokens regardless of the network, I would say we do not need this check.

If Mary and Artem say the user select tokens for the specific network, then instead of not displaying the dropdown why not display it either as a paragraph (label+value) or disabling the dropdown that way we provide the user a feedback to the user about which network will be use.

But, I would still tend to allowing user to always administrate their all tokens regardless of the network.

Reply:
According to Artem, we should show filter by network, but I agree with you to leave the choice. The only thing is that it will default to the selected network is any, right?

Reply:
I would do:

- If the modal is open on "Chain Fusion", then the default value is `undefined`.
- If the modal is open with a selected network, then the default is the network and the value cannot be changed

Reply:
cool!